lib/decompress/zstd_decompress.rs

```diff
@@ -1074,7 +1074,10 @@ fn get_frame_header_advanced(zfhPtr: &mut ZSTD_FrameHeader, src: &[u8], format:
     }
 
     // ensure there is enough `src` to fully read/decode frame header
-    let fhsize = frame_header_size_internal(src, format);
+    let fhsize = match frame_header_size_internal(src, format) {
+        Ok(size) => size,
+        Err(err) => err.to_error_code(),
+    };
```

Comment:
i _think_ that a `?` would also be correct here. The errors have very high values, so the check below will fail for errors. Would need some fuzzing though.